doc/rtd/howto/status.rst

```diff
@@ -58,5 +58,29 @@ See the list of all possible reported statuses:
     "degraded running"
     "disabled"
 
+Cloud-init enablement status
+----------------------------
+
+Separate from the current running status described above, cloud-init can also
+report how it was disabled or enabled. This can be viewed by checking
+the `boot_status_code` in ``cloud-init status --long``, which may
+contain any of the following states:
+
+- ``'unknown'``: ``ds-identify`` hasen't run yet to determine if cloud-init should
+  be run during this boot
+- ``'disabled-by-marker-file'``: /etc/cloud/cloud-init.disabled exists
+  which prevents cloud-init from ever running
+- ``'disabled-by-generator'``: ``ds-identify`` determined no applicable cloud-init
+  datasources
+- ``'disabled-by-kernel-cmdline'``: kernel cmdline contained
+  cloud-init=disabled
+- ``'disabled-by-environment-variable'``: environment variable KERNEL_CMDLINE
+  contained cloud-init=disabled
+- ``'enabled-by-kernel-cmdline'``: kernel cmdline contained
+  cloud-init=enabled
+- ``'enabled-by-generator'``: ``ds-identify`` detected possible cloud-init
+  datasources
+- ``'enabled-by-sysvinit'``: enabled by default in SysV init environment
```

Comment:
```suggestion
- ``'unknown'``: ``ds-identify`` has not run yet to determine if cloud-init should
  be run during this boot
- ``'disabled-by-marker-file'``: :file:`/etc/cloud/cloud-init.disabled` exists
  which prevents cloud-init from ever running
- ``'disabled-by-generator'``: ``ds-identify`` determined no applicable cloud-init
  datasources
- ``'disabled-by-kernel-cmdline'``: environment variable ``KERNEL_CMDLINE`` contained
  ``cloud-init=disabled``
- ``'disabled-by-environment-variable'``: environment variable ``KERNEL_CMDLINE``
  contained ``cloud-init=disabled``
- ``'enabled-by-kernel-cmdline'``: ``KERNEL_CMDLINE`` contained
  ``cloud-init=enabled``
- ``'enabled-by-generator'``: ``ds-identify`` detected possible cloud-init
  datasources
- ``'enabled-by-sysvinit'``: enabled by default in SysV init environment
```

Reply:
One exception here: "``KERNEL_CMDLINE`` contained ``cloud-init=enabled``" is not right. In this case we are talking literally about the kernel command line. The line above is referring to an environment variable with the name "KERNEL_CMDLINE". The kernel commandline itself can be read via `cat /proc/cmdline` and is different from the aforementioned environment variable with a similar name.

Reply:
Thanks for the context - I saw the discrepancy and figured they were all meant to be the same.

Reply:
(to disambiguate, I would probably propose changing cmdline to command line, but I leave that to your discretion - happy to approve either way)

Reply:
agreed, looks like we can drop commandline from the spelling list too